content/client-lib-development-guide/versioning.textile

```diff
@@ -0,0 +1,17 @@
+---
+title: Ably Versioning
+section: client-lib-development-guide
+index: 19
+---
+
+Ably aims to follow the "semantic versioning scheme":http://semver.org/ for our "client library SDKs":https://www.ably.io/download, the "realtime protocol used by these SDKs":/client-lib-development-guide/protocol, our "REST API endpoints":/rest-api, and all other public endpoints, that support versioning, such as "Server-Sent Events":/sse.
+
+Whilst we recognise this approach has some limitations, primarily around the need to synchronize version updates across all of the above when a breaking change is introduced, we believe that having an independent series of version numbers for different protocols, endpoints and SDKs, would make managing that incredibly difficult for Ably, and most importantly, for our customers. This documentation itself operates on the basis that customer facing content shares a single version number so that customers can view the latest, or, for example, switch to 1.1 across the entire site (where content has been versioned).
+
+Further, given that the version numbers we use cover the raw HTTP API, the protocol and API spec, and the client library SDKs, features such as support for connection/request params, channel params, extras, token lengths, APIs (eg push HTTP API) vary by the API version. Stateless connections are subject to some of those constraints, even if they don't follow the full protocol spec or the library API.
+
+As such, in the case of SSE for example, we feel standardising on versioning means it's quite natural for a connection string for a stateless connection to include the spec version, and that will allow the system to know what features the client can be expected to understand, and might be required so that the system knows how to interpret param values that are supplied by the client. Whilst we always try to make it so that there are no incompatibilities, we have a version because we recognise that that's not always avoidable.
```

Comment:
TBH I'm not sure I'm 100% convinced - the main api version is essentially 100% driven by client library protocol requirements, so this policy leaves other transports wanting to make changes that don't affect client libs in the position of just waiting until the next new client lib spec version happens, which might be a month or > a year away. (And technically per semver if it's a breaking change, waiting until the next new _major_ client lib spec version happens, which is once in a blue moon).

But I don't feel particularly strongly about it, and if you (Matt) and Paddy both think this is the right policy, I'm happy to accept the consensus.

Reply:
I agree it's not ideal, but equally maintaining different versions across different protocols, SDKs, endpoints feels like a far bigger issue.  If we had to introduce a breaking change that only affected SSE for example, and we bumped from 1.3 to 1.4 but communicate it in the changelogs, it would be less than ideal, but possible.  Equally, I think we can deal with that issue if it ever arises.